R/finalsize_risk_grps.R

```diff
@@ -84,18 +86,30 @@ final_size_grps <- function(contact_matrix,
     susceptibility = susceptibility
   )
 
-  # solve for final size
-  epi_final_size <- solve_final_size_iterative(
-    contact_matrix = epi_spread_data[["contact_matrix"]],
-    demography_vector = epi_spread_data[["demography_vector"]],
-    p_susceptibility = epi_spread_data[["p_susceptibility"]],
-    susceptibility = epi_spread_data[["susceptibility"]],
-    iterations = iterations,
-    tolerance = tolerance,
-    step_rate = step_rate,
-    adapt_step = adapt_step
-  )
-
+  epi_final_size <- numeric()
+  # solve for final size using iterative solver
+  if (solver == "iterative") {
+    epi_final_size <- solve_final_size_iterative(
+      contact_matrix = epi_spread_data[["contact_matrix"]],
+      demography_vector = epi_spread_data[["demography_vector"]],
+      p_susceptibility = epi_spread_data[["p_susceptibility"]],
+      susceptibility = epi_spread_data[["susceptibility"]],
+      iterations = iterations,
+      tolerance = tolerance,
+      step_rate = step_rate,
+      adapt_step = adapt_step
+    )
+  } else if (solver == "newton") {
+    epi_final_size <- solve_final_size_newton(
+      contact_matrix = epi_spread_data[["contact_matrix"]],
+      demography_vector = epi_spread_data[["demography_vector"]],
+      susceptibility = epi_spread_data[["susceptibility"]],
+      iterations = iterations,
+      tolerance = tolerance
+    )
```

Comment:
If / when the arguments to the solver are passed as `control`, this could become:

```
f_solver <- switch(solver,
  iterative = solve_final_size_iterative,
  newton = solve_final_size_newton
)
epi_final_size <- solve_final_size_newton(
  contact_matrix = epi_spread_data[["contact_matrix"]],
  demography_vector = epi_spread_data[["demography_vector"]],
  susceptibility = epi_spread_data[["susceptibility"]],
  control = control
)
```

to remove redundancy in the code and simplify a potential future renaming of the arguments.

Reply:
Done in https://github.com/epiverse-trace/finalsize/pull/65/commits/891ed778bc79ab9e8d5697b5feee6da02fd720f0